src/components/SignUpForm.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from "react";
+import { Form, Input, Container } from "semantic-ui-react";
+import { auth } from "../modules/auth";
+import { useHistory } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+
+const SignUpForm = () => {
+  const signupMessage = useSelector(state => state.messages.signupMessage);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  useEffect(() => {
+    dispatch({ type: "RESET_ACTIVE_PAGE" });
+  }, [dispatch]);
+
+  const signup = async (e) => {
+    try {
+      const response = await auth.signUp({
+        email: e.target.email.value,
+        password: e.target.password.value,
+        password_confirmation: e.target.passwordConfirmation.value,
+      });
+      if (response.data.status === "success") {
+        history.replace("/login");
+        dispatch({
+          type: "SET_SIGNUP_MESSAGE",
+          payload: { signupMessage: response.data.message },
+        });
+      }
+    } catch (error) {
+      dispatch({
+        type: "SET_SIGNUP_MESSAGE",
+        payload: { signupMessage: error.response.data.errors.full_messages },
```

Comment:
is this coming from newsroom challenge ? error.response.data.errors.full_messages

Reply:
yes but it comes from a function so I used that as well